consoles/console.pm

```diff
@@ -85,6 +86,29 @@ sub select {
     return $activated;
 }
 
+sub save_snapshot {
+    my ($self, $name) = @_;
+    $self->snapshot($name, 'activated', $self->{activated});
+}
+
+sub load_snapshot {
+    my ($self, $name) = @_;
+    $self->{activated} = $self->snapshot($name, 'activated') // 0;
+}
+
+sub snapshot {
+    my ($self, $name, $key, $value) = @_;
+    return undef unless defined($name);
+
+    if (defined($key) && defined($value)) {
+        $self->{snapshots}->{$name} //= {};
+        $self->{snapshots}->{$name}->{$key} = $value;
+    } elsif (defined($key)) {
+        return ($self->{snapshots}->{$name}) ? $self->{snapshots}->{$name}->{$key} : undef;
```

Comment:
Shouldn't we check for existence of `$name` in snapshots in the elsif branch?

Reply:
its in line 101, right?

Reply:
That is only whether the argument is `undef`, but the key might still be missing in snapshots sub hash.

Reply:
Sorry I misunderstood you. I check `$self->{snapshots}->{$name}` in that ternary operator.

Reply:
ok, so explicit test is not needed. https://perldoc.perl.org/functions/exists#exists-EXPR  thanks